lms/djangoapps/courseware/access.py

```diff
@@ -687,3 +688,11 @@ def get_user_role(user, course_key):
         return 'staff'
     else:
         return 'student'
+
+
+def in_preview_mode():
+    """
+    Returns whether the user in preview mode or not.
+    """
+    hostname = get_current_request_hostname()
+    return hostname and 'preview' in hostname
```

Comment:
I think it would be safer to split the hostname on "." character and check if one of the values exactly matches 'preview'.